### DISCUSSION TITLE
API C# Entity - GetIssuedDocumentResponse

### DISCUSSION THREAD

**Discussion by Mustino2022** [fattureincloud/api#153] [Q&A] [answered]
Buongiorno
nella documentazione API Reference sono disponibili first_name, last_name, email, type
![image](https://user-images.githubusercontent.com/117667863/200783839-40183d42-5eec-4e71-a33e-2e7c7ab5ce79.png)

Mentre nella risposta JSON reale le proprietà non è sono disponibili
![image](https://user-images.githubusercontent.com/117667863/200784908-ed2ec284-ec87-4363-b863-0e6d553010b6.png)

in c# le proprietà ci sono ma risultano null
![image](https://user-images.githubusercontent.com/117667863/200787768-0be5f8af-aef6-4315-a774-06f1a4436ec0.png)

![image](https://user-images.githubusercontent.com/117667863/200785626-a7acfefc-a522-4991-ac6a-8d136bb240a8.png)

Se può servire l'id del documento in oggetto è 241419445

Restando in attesa di riscontro, ringrazio e saluto cordialmente

**Comment by valmoz** [comment, marked as answer]
Buongiorno,
per avere accesso ai dettagli completi del cliente suggeriamo di utilizzare il metodo GetClient o GetSupplier.

Il campo entity di Issued Document infatti può rappresentare entrambi questi tipi, e per questo motivo abbiamo utilizzato un modello unificato nella rappresentazione OpenAPI.

- [GetClient](https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/entities/clients/-client_id-)
- [GetSupplier](https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/entities/suppliers/-supplier_id-)
- [OpenAPI Specification](https://developers.fattureincloud.it/docs/openapi-specification)

**Reply by Mustino2022** [reply]
Buongiorno @valmoz , grazie per la risposta, il problema è che in fase di creazione fattura, non creo il cliente ma salvo i suoi dati direttamente in fattura.
Per accedere a tutte le informazioni della [GetClient] immagino dovrò salvare il cliente? E' corretto?

**Reply by valmoz** [reply]
Esattamente. 
Consideri poi che nel caso di cliente inserito direttamente in fattura e non in anagrafica noi ignoriamo tutti i parametri che non vengono effettivamente utilizzati in fattura (in quanto non verrebbero in ogni caso memorizzati), quindi se vuole avere a disposizione tutti i parametri è indispensabile effettuare l'aggiornamento dell'anagrafica.
